cloudify_rest_client/aria/mapi/mapi.py

```diff
@@ -0,0 +1,123 @@
+        # `close` method per mapi instead.
+        self._session = type(
+            'MockSession', (object,), {'remove': lambda *a, **kw: None})()
+        self._engine = type(
+            'MockSession', (object,), {'dispose': lambda *a, **kw: None})()
+
+    def put(self, entry, **kwargs):
+        response = requests.put(
+            '/'.join([self._endpoint, self.name]),
+            data=pickle.dumps(entry._obj
+                              if isinstance(entry, wrappers.WrapperBase)
+                              else entry)
+        )
+        return self._respond(response, self.name)
+
+    def create(self, **kwargs):
+        pass
+
+    def get(self, entry_id, **kwargs):
+        suffix = '/'.join([self.name, str(entry_id)])
+        response = requests.get('/'.join([self._endpoint, suffix]))
+        return self._respond(response, suffix)
+
+    def refresh(self, entry):
+        return self.get(entry.id)
+
+    def iter(self, **kwargs):
+        response = requests.get('/'.join([self._endpoint, self.name]),
+                                json=kwargs)
+        for item in self._respond(response, self.name):
+            yield item
+
+    def update(self, entry, **kwargs):
+        suffix = '/'.join([self.name, str(entry.id)])
+        response = requests.patch('/'.join([self._endpoint, suffix]),
+                                  data=pickle.dumps(entry._obj))
+        return self._respond(response, suffix)
+
+    def delete(self, entry_id, **kwargs):
+        pass
+
+    def _respond(self, response, attribute_path):
+        try:
+            return self._wrap(pickle.loads(response.json()), attribute_path)
+        except Exception:
+            raise exceptions.RequestException(response.text)
+
+    def _wrap(self, value, attribute_path):
+        from aria.modeling.mixins import ModelMixin
+
+        wrapper = None
+        kw = {}
+
+        if isinstance(value, dict):
+            wrapper = wrappers.DictWrapper
+            kw = dict(
+                    (key, self._wrap(value, '/'.join([attribute_path, key])))
+                    for key, value in (value or {}).items()
+                )
+        elif isinstance(value, list):
+            wrapper = wrappers.ListWrapper
+            seq = [self._wrap(
+                item, '/'.join([attribute_path, str(getattr(item, 'id', i))])
+            ) for i, item in enumerate(value or [])]
+            kw = dict(seq=seq)
+
+        elif isinstance(value, ModelMixin):
+            wrapper = wrappers.ObjectWrapper
+            kw = dict(obj=value)
+
+        if wrapper is not None and callable(wrapper):
+            return wrapper(
+                _attribute_path=attribute_path,
+                _query=partial(self._get_query,
+                               end_point=self._endpoint,
+                               responder=self._respond),
+                **kw
+            )
+
+        return value
+
+    @classmethod
+    def _get_query(cls, entry_path, end_point, responder):
```

Comment:
explain what is the entry path (what is its relation to the attribute path, etc).